pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -692,29 +692,24 @@ func (aws *AWSCloud) NodeAddresses(name string) ([]api.NodeAddress, error) {
 }
 
 // ExternalID returns the cloud provider ID of the specified instance (deprecated).
-// Note that if the instance does not exist or is no longer running, we must return ("", cloudprovider.InstanceNotFound)
+// We assume that if this is run on the instance itself, the instance exists and is alive
 func (aws *AWSCloud) ExternalID(name string) (string, error) {
-	// We must verify that the instance still exists
-	instance, err := aws.findInstanceByNodeName(name)
+	awsInstance, err := aws.getSelfAWSInstance()
 	if err != nil {
 		return "", err
 	}
-	if instance == nil || !isAlive(instance) {
-		return "", cloudprovider.InstanceNotFound
-	}
-	return orEmpty(instance.InstanceId), nil
+	return awsInstance.awsID, nil
 }
 
 // InstanceID returns the cloud provider ID of the specified instance.
 func (aws *AWSCloud) InstanceID(name string) (string, error) {
-	// TODO: Do we need to verify it exists, or can we just construct it knowing our AZ (or via caching?)
-	inst, err := aws.getInstanceByNodeName(name)
+	awsInstance, err := aws.getSelfAWSInstance()
```

Comment:
Same comment as ExternalID: we should probably keep the ability to query by name